src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -33,7 +33,16 @@ namespace Dynamo.ViewModels
 {
     public interface IDynamoViewModel : INotifyPropertyChanged
     {
-        ObservableCollection<WorkspaceViewModel> Workspaces { get; set; } 
+        ObservableCollection<WorkspaceViewModel> Workspaces { get; set; }
+
+        /// <summary>
+        /// Executes node and connector creation commands to create node1 and connect it to node2
+        /// </summary>
+        /// <param name="node1">input node to create</param>
+        /// <param name="node2">node to connect to</param>
+        /// <param name="portIndex1">output port of input node</param>
+        /// <param name="portIndex2">input node of given node</param>
+        void CreateAndConnectInputNode(NodeModel node1, NodeModel node2, int portIndex1, int portIndex2);
```

Comment:
Sorry to make this such a long explanation, but I'm trying (failing) to make sure that the `IExtension` framework API is utterly stable - this means we can move code out of `DynamoCore` and have a more maintainable code base - but this requires we really get the design right.